include/boost/json/value.hpp

```diff
@@ -2832,6 +2832,65 @@ struct key_value_pair
 
 //----------------------------------------------------------
 
+#ifdef BOOST_JSON_DOCS
+
+/** Tuple-like element access.
+    
+    This overload permits the key and value
+    of a `key_value_pair` to be accessed
+    by index. For example:
+
+    @code 
+
+    key_value_pair kvp("num", 42);
+    
+    auto key = get<0>(kvp);
+    auto value = get<1>(kvp);
+
+    @endcode
+
+    In addition to calling the function directly,
+    this function allows `key_value_pair` objects
+    to initialize structured bindings:
+
+    @code
+
+    key_value_pair kvp("num", 42);
+
+    auto& [key, value] = kvp;
+
+    @endcode
+
+    The return type of this template
+    depends upon the value of `I`.
+    If `I` is 0, the return type is `const string_view`.
+    If `I` is 1, the return type is `value&` if
+    the argument is a non-const lvalue, `value&&` if 
+    the argument is a non-const rvalue, and 
+    `const value&` otherwise..
+    Any other value for `I` is ill-formed.
+
+    @tparam I The element index to access.
+
+    @par Constraints
+
+    `T` is (possibly cv- or reference-qualified)
+    `key_value_pair`.
```

Comment:
Use `@ref key_value_pair` and it will link

Reply:
Actually the convention in the docs (I think) is to use a C++ expression here. So we might write something like:
```
@par Constraints

`std::is_same< std::remove_cvref_t<T>, key_value_pair >`
```